contracts/transfer/src/state.rs

```diff
@@ -269,21 +272,26 @@ impl TransferState {
         true
     }
 
-    /// Spend the inputs and process the outputs, together with the crossover.
-    /// It performs all checks necessary to ensure the transaction is valid -
-    /// hash matches, anchor has been a root of the tree, proof checks out,
-    /// etc...
+    /// Spend the inputs and process the outputs, together with the crossover,
+    /// and executes the contract call if present. It performs all checks
+    /// necessary to ensure the transaction is valid - hash matches, anchor
+    /// has been a root of the tree, proof checks out, etc...
     ///
     /// This will emplace the crossover in the state, if it exists, and expect
     /// [`refund`] to be called if it succeeds.
     ///
+    /// If a contract call is present, it will be executed as the last step.
```

Comment:
I don't understand this comment. Do you mean any contract call that is present in `tx` will be executed after `spend_and_execute`?

Reply:
I wanted to say it was executed as the last step of this function. Was admittedly a bit unclear. Fixed